src/suricata.c

```diff
@@ -460,13 +460,18 @@ static int SetBpfString(int argc, char *argv[])
     tmpindex = optind;
     while(argv[tmpindex] != NULL) {
         strlcat(bpf_filter, argv[tmpindex],bpf_len);
-        if(argv[tmpindex + 1] != NULL) {
+	if (*bpf_filter == '-') {
```

Comment:
Indentation is off (tab ?).

Reply:
I don't like to get error handling of parsing in the initialization code.

Reply:
Okay, I will see about reworking this with your suggestions, thanks @regit